### ISSUE TITLE
Keyboard navigation toolbox has runtime failure when opening 'loops'

### ISSUE THREAD

**Issue by BenHenning** [google/blockly#8979]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

When running the latest keyboard navigation plugin linked against the latest v12, trying to open the 'loops' section in the plugin's test playground causes a runtime error.

### Reproduction steps

1. Check out 1c79e1ed77e5a341abbb01a5807abe71079a2297 in Core.
2. Check out 5378c9894f8c6df3ebf3be68a826706783d26c35 in the navigation plugin.
3. Build a release version of Core and npm link it to the plugin.
4. Run the plugin's test server.
5. Open the plugin's test playground and try to open 'loops'.
6. Observe the runtime failure.

### Stack trace

```
Block not present in workspace's list of top-most blocks.
    at WorkspaceSvg$$module$build$src$core$workspace_svg.removeTopBlock (webpack-internal:///../blockly/dist/blockly_compressed.js:1296:10)
    at WorkspaceSvg$$module$build$src$core$workspace_svg.removeTopBlock (webpack-internal:///../blockly/dist/blockly_compressed.js:1354:7)
    at BlockSvg$$module$build$src$core$block_svg.dispose (webpack-internal:///../blockly/dist/blockly_compressed.js:927:218)
    at BlockSvg$$module$build$src$core$block_svg.dispose (webpack-internal:///../blockly/dist/blockly_compressed.js:1006:353)
    at BlockFlyoutInflater$$module$build$src$core$block_flyout_inflater.disposeItem (webpack-internal:///../blockly/dist/blockly_compressed.js:1629:31)
    at eval (webpack-internal:///../blockly/dist/blockly_compressed.js:1052:258)
    at Array.forEach (<anonymous>)
    at VerticalFlyout$$module$build$src$core$flyout_vertical.clearOldBlocks (webpack-internal:///../blockly/dist/blockly_compressed.js:1052:190)
    at VerticalFlyout$$module$build$src$core$flyout_vertical.show (webpack-internal:///../blockly/dist/blockly_compressed.js:1048:425)
    at Toolbox$$module$build$src$core$toolbox$toolbox.refreshSelection (webpack-internal:///../blockly/dist/blockly_compressed.js:1751:164)
```

Sometimes there's also a different error (this happened when I tried opening 'Lists'):

```
Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at String.replace (<anonymous>)
    at FieldNumber$$module$build$src$core$field_number.doClassValidation_ (webpack-internal:///../blockly/dist/blockly_compressed.js:1546:367)
    at FieldNumber$$module$build$src$core$field_number.setValue (webpack-internal:///../blockly/dist/blockly_compressed.js:732:18)
    at new FieldNumber$$module$build$src$core$field_number (webpack-internal:///../blockly/dist/blockly_compressed.js:1544:921)
    at FieldNumber$$module$build$src$core$field_number.fromJson (webpack-internal:///../blockly/dist/blockly_compressed.js:1548:175)
    at Object.fromJsonInternal$$module$build$src$core$field_registry [as fromJsonInternal] (webpack-internal:///../blockly/dist/blockly_compressed.js:336:180)
    at $.fromJson$$module$build$src$core$field_registry (webpack-internal:///../blockly/dist/blockly_compressed.js:335:268)
    at BlockSvg$$module$build$src$core$block_svg.fieldFromJson (webpack-internal:///../blockly/dist/blockly_compressed.js:956:380)
    at BlockSvg$$module$build$src$core$block_svg.interpolate (webpack-internal:///../blockly/dist/blockly_compressed.js:955:100)
    at BlockSvg$$module$build$src$core$block_svg.jsonInit (webpack-internal:///../blockly/dist/blockly_compressed.js:951:42)
```

**Comment by BenHenning** [comment]
Quick investigation suggests that this started happening with the merge of #8946 FYI @cpcallen.
